server/monitor/subscriptions.go

```diff
@@ -178,16 +178,30 @@ func (m *Monitor) CreateAndSaveChatSubscription(mmSubscription *storemodels.Glob
 }
 
 func (m *Monitor) recreateChatSubscription(subscriptionID, userID, secret string) error {
-	if err := m.client.DeleteSubscription(subscriptionID); err != nil {
-		m.api.LogDebug("Unable to delete old subscription, maybe it doesn't exist anymore in the server", "error", err.Error())
+	var err error
+	if subscriptionID != "" {
+		if err = m.client.DeleteSubscription(subscriptionID); err != nil {
+			m.api.LogDebug("Unable to delete old subscription, maybe it doesn't exist anymore in the server", "error", err.Error())
+		}
 	}
 
 	newSubscription, err := m.client.SubscribeToUserChats(userID, m.baseURL, m.webhookSecret, !m.useEvaluationAPI, m.certificate)
 	if err != nil {
 		return err
 	}
 
-	return m.store.SaveChatSubscription(storemodels.ChatSubscription{SubscriptionID: newSubscription.ID, UserID: userID, Secret: secret, ExpiresOn: newSubscription.ExpiresOn, Certificate: m.certificate})
+	if subscriptionID == "" {
```

Comment:
I don't understand this new check: if I pass `subscriptionID != ""` then we delete the old one and don't save the new one?

Reply:
Yes, that's wrong, I don't know if was a brain fart or a merge problem, but clearly the idea is not that.

Reply:
ohh, no, wait, what is going on here, is that we are deleting the "MS Teams subscription" and creating it again, but the local stored subscription is created brand new or updated based on the existing one. To ensure we preserve things like the last update at, and things like that.